### PR TITLE
Bratseth/document get

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: client/go/util/operationResult.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// A struct containing the result of an operation
+// Author: bratseth
+
+package util
```
Comment (mpolden) [inline client/go/util/operationResult.go:5]: Filenames should use snake case: operationResult.go -> operation_result.go.

File: client/go/vespa/document.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	} else {
+		return util.FailureWithPayload(serviceDescription+" at "+request.URL.Host+": "+response.Status, util.ReaderToJSON(response.Body))
+	}
+}
+
+func Post(documentId string, jsonFile string, target string) *util.OperationResult {
+	header := http.Header{}
+	header.Add("Content-Type", "application/json")
+
+	fileReader, fileError := os.Open(jsonFile)
```
Comment (mpolden) [inline client/go/vespa/document.go:54]: This should be closed: `defer fileReader.Close()` after checking the error.

File: client/go/util/operationResult.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+type OperationResult struct {
+	Success bool
+	Message string // Mandatory message
+	Detail  string // Optional detail message
+	Payload string // Optional payload - may be present whether or not the operation was success
+}
+
+func Success(message string) *OperationResult {
```
Comment (mpolden) [inline client/go/util/operationResult.go:14]: I think it's better to return a copy here (and the other constructors), seeing as this a light-weight struct.

File: client/go/util/operationResult.go (hunk 1 above)
Comment (bratseth) [reply client/go/util/operationResult.go]: I couldn't find any consensus on this but ok.

File: client/go/vespa/document.go (hunk 2 above)
Comment (bratseth) [reply client/go/vespa/document.go:59]: Thanks!

**Pull request merged**